Add FAQ for No results error for log search (#6294)
SumoLogic/sumologic-documentation: docs/search/faq.md

### Source expression issues

Source expression is the part of the query which comes before the first pipe (|), that defines which logs are searched in the first place. A wrong source expression returns no results because the scope does not match any ingested logs, the search engine has no data to process, or the query returns an empty result set even before filters or operators are applied. To avoid this issue, make sure you satisfy the following conditions:

- Metadata field values (for example, `_sourceCategory`, `_sourceHost`, `_collector`) are correct and not misspelled.
- The specified log source or category exists.
- The selected time range contains logs matching the source expression.
- You have access to the required log sources.

### Query filters and operator issues

> Filters and operators (for example, `where`, `filter`, `parse`, `stats`) progressively refine the dataset returned by the source expression. Each filter or operator keeps only the records that meet its specified condition. If none of the records satisfy a condition at any stage of the query pipeline, the dataset becomes empty and the search returns no results. To avoid this issue, make sure you satisfy the below conditions:

Author: ```suggestion
Filters and operators (for example, `where`, `filter`, `parse`) progressively refine the dataset returned by the source expression. Each filter or operator keeps only the records that meet its specified condition. If none of the records satisfy a condition at any stage of the query pipeline, the dataset becomes empty and the search returns no results. To avoid this issue, make sure you satisfy the below conditions:
```

- Filter conditions are not overly restrictive.
- Correct field names and parsed fields are used in comparisons and filter conditions.
- Parsing rules execute successfully and produce the fields referenced later in the query.

> - Aggregation or transformation operators (such as `stats`, `timeslice`, or `top`) are not applied to an already filtered-out dataset.

Author: ```suggestion
- Aggregation or transformation operators (such as `timeslice`, or `top`) are not applied to an already filtered-out dataset.
```

### Access permission issues (role-based search filters)

> Role-based access controls can also result in empty search results as they limit which log data is allowed for you to view. Each user role may include search filters that automatically restrict which log sources can be queried. If your role does not permit access to the requested sources, those logs are excluded from the search results. To avoid this issue, make sure you satiety the below conditions:

Author: @AyanGhatak we can do that.

> Role-based access controls can also result in empty search results as they limit which log data is allowed for you to view. Each user role may include search filters that automatically restrict which log sources can be queried. If your role does not permit access to the requested sources, those logs are excluded from the search results. To avoid this issue, make sure you satiety the below conditions:

Author: ```suggestion
[Role-based access controls](/docs/manage/users-roles/roles/role-based-access-control/) can also result in empty search results as they limit which log data is allowed for you to view. Each user role may include search filters that automatically restrict which log sources can be queried. If your role does not permit access to the requested sources, those logs are excluded from the search results. To avoid this issue, make sure you satisfy the below conditions:
```
